core/src/avm2/globals.rs

```diff
@@ -1029,3 +1030,85 @@ pub fn load_player_globals<'gc>(
 
     Ok(())
 }
+
+/// This file is built by 'core/src/avm2/playerglobal/compile_swc.sh'
+/// See that script, and 'core/src/avm2/playerglobal/README.md', for more details
+const PLAYERGLOBAL: &[u8] = include_bytes!("playerglobal/library.swf");
+
+/// Loads classes from our custom 'playerglobal' (which are written in ActionScript)
+/// into the environment. See 'core/src/playerglobal/README.md' for more information
+fn load_playerglobal<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    domain: Domain<'gc>,
+) -> Result<(), Error> {
+    let movie = Arc::new(SwfMovie::from_data(PLAYERGLOBAL, None, None)?);
+
+    let slice = SwfSlice::from(movie);
+
+    let mut reader = slice.read_from(0);
+
+    let tag_callback = |reader: &mut SwfStream<'_>, tag_code, tag_len| {
+        if tag_code == TagCode::DoAbc {
+            let start = reader.as_slice();
+            // TODO: The tag reader parses the entire ABC file, instead of just
+            // giving us a `SwfSlice` for later parsing, so we have to replcate the
+            // *entire* parsing code here. This sucks.
+            let flags = reader.read_u32()?;
+            let name = reader.read_str()?.to_string_lossy(reader.encoding());
+            let is_lazy_initialize = flags & 1 != 0;
+            let num_read = reader.pos(start);
+
+            let remaining_len = tag_len - num_read;
+
+            let slice = slice
+                .resize_to_reader(reader, remaining_len)
+                .ok_or_else(|| {
+                    std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        "Invalid source or tag length when running init action",
+                    )
+                })?;
+
+            Avm2::load_abc(
```

Comment:
Why not embedding just one `DoAction` tag, without a whole SWF? Looks like it can reduce some complexity.

Reply:
I think it's better to keep `playerglobals.swf` in a (sort of) standard format, so that JPEXS can read it. If we were to concatenate all of the classes into a single `DoAction` tag (rather than a `DoAbc` per class), I think we'd lose that ability.

Reply:
Understood. But in the cost/benefit balance I still see it much more efficient to embed `playerglobal.abc` over `playerglobal.swc`. Here are few examples:
* Flex SDK supports output of `.abc` files (`asc` does this by default). With this in use, `prepare_playerglobal` script won't be necessary, and we get reduced complexity.
* Loading an `.abc` file should have better performace than `.swc`, resulting in improved load times of Ruffle.
* As as subslice of `.swc`, `.abc` files are trivially smaller, and need less code for being loaded, which is good for Ruffle's binary size and its code complexity.

On the other hand, unable to open `playerglobal.abc` in JPEXS might be a cumbersome, but I expect it to be very rare to actually be needed. And even if it is needed, `compile_swc.sh` can be hacked to output a `.swc` instead.

Reply:
> Flex SDK supports output of .abc files (asc does this by default). With this in use, prepare_playerglobal script won't be necessary, and we get reduced complexity.

We would still need to merge those ABC files into one, which would still require invoking some external tool.

> Loading an .abc file should have better performace than .swc, resulting in improved load times of Ruffle.

The `playerglobal.swf` file that we build only contains `DoAbc` tags - so the only overhead is the SWF header, and the tag code + length for each `DoAbc` tag. I think this is a minuscule amount of overhead to remove, especially considering how useful I found JPEXS when working on this PR.

> As as subslice of .swc, .abc files are trivially smaller, and need less code for being loaded, which is good for Ruffle's binary size and its code complexity.

Note that by using a `.abc` file, we would lose out on the compression supported by the `.swf` format - unless we wanted to do something even *more* nonstandard, and compress that `.abc` file itself.

I think it's premature to move away from the standard `.swf` format, especially when we only have two classes in `playergloal`. If load times become an issue in the future, we can re-evaluate how we implement `playerglobals` - but I think we should preserve JPEXS compatibility for as long as we reasonably can.